drivers/overlay/peerdb.go

```diff
@@ -301,6 +308,14 @@ func (d *driver) peerAdd(nid, eid string, peerIP net.IP, peerIPMask net.IPMask,
 		return fmt.Errorf("could not add neighbor entry into the sandbox: %v", err)
 	}
 
+	if existsErr == nil && vtep.String() != oldVtep.String() &&
+		oldVtep.String() != d.advertiseAddress {
+		if err := sbox.DeleteNeighbor(oldVtep, oldMac, true); err != nil {
```

Comment:
like what is fetched at line 129 can be already different from what is configured in the overlay namespace